src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -19372,49 +19413,15 @@ namespace System
 }
 ";
             var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseExe);
-            var verifier = CompileAndVerify(comp, expectedOutput: " 111 112 113 114", verify: Verification.Fails).VerifyDiagnostics();
-
-            verifier.VerifyIL("Program.Test",
-@"
-{
-  // Code size       56 (0x38)
-  .maxstack  2
-  .locals init (System.Collections.Generic.IEnumerator<int> V_0,
-                Buffer4<int> V_1)
-  IL_0000:  ldarg.0
-  IL_0001:  ldfld      ""Buffer4<int> C.F""
-  IL_0006:  stloc.1
-  IL_0007:  ldloca.s   V_1
-  IL_0009:  call       ""System.Collections.Generic.IEnumerator<int> Buffer4<int>.GetEnumerator()""
-  IL_000e:  stloc.0
-  .try
-  {
-    IL_000f:  br.s       IL_0023
-    IL_0011:  ldloc.0
-    IL_0012:  callvirt   ""int System.Collections.Generic.IEnumerator<int>.Current.get""
-    IL_0017:  ldc.i4.s   32
-    IL_0019:  call       ""void System.Console.Write(char)""
-    IL_001e:  call       ""void System.Console.Write(int)""
-    IL_0023:  ldloc.0
-    IL_0024:  callvirt   ""bool System.Collections.IEnumerator.MoveNext()""
-    IL_0029:  brtrue.s   IL_0011
-    IL_002b:  leave.s    IL_0037
-  }
-  finally
-  {
-    IL_002d:  ldloc.0
-    IL_002e:  brfalse.s  IL_0036
-    IL_0030:  ldloc.0
-    IL_0031:  callvirt   ""void System.IDisposable.Dispose()""
-    IL_0036:  endfinally
-  }
-  IL_0037:  ret
-}
-");
+            comp.VerifyDiagnostics(
+                // (21,27): error CS9185: foreach statement on an inline array of type 'Buffer4<int>' is not supported
+                //         foreach (var y in x.F)
+                Diagnostic(ErrorCode.ERR_InlineArrayForEachNotSupported, "x.F").WithArguments("Buffer4<int>").WithLocation(21, 27)
```

Comment:
>ERR_InlineArrayForEachNotSupported [](http://example.com/codeflow?start=37&length=34)

Why is this case no longer supported?

Reply:
> Why is this case no longer supported?

I do not think it is worth supporting fall back for scenarios with malformed span types. At least from the start. The spec says foreaching over an inline array is like foreaching over a span. The span type is malformed in this case.

> Were we falling through to a non-inline array case in ForEachLoopBinder?

Yes.